### DISCUSSION TITLE
is it possible to reach a variable / function from another page ?

### DISCUSSION THREAD

**Discussion by BilalKhatib** [vuejs/petite-vue#189] [Q&A] [answered]
### I have 2 pages : Page1.html and Page2.html

**Page1.html** has a declared variable and function in it 

````````````````````````````````````````````````````````````````````````````````````````````````````````````````````````````````````````````````````````````````````````````````````````````````
username : 'admin',

welcomeUser() {
     console.log('Welcome Back  ' + this.username)
},
````````````````````````````````````````````````````````````````````````````````````````````````````````````````````````````````````````````````````````````````````````````````````````````````

i need to access those function and variable in my Page2.html createApp
Any help ?

**Comment by sqllyw** [comment, marked as answer]
I use this in page1:

     window.localStorage.setItem(key, value);

in page2:

    let state = window.localStorage.getItem(key)

**Reply by sqllyw** [reply]
I'll assume your app is multi pages, another approach might be using query string, 

in page1, 
````
const url = new URL("page2");
url.searchParams.set("username", 'admin'); 
location.href = url.href   
````
then you retrieve the 'username' in page2 before createApp

if you want the passing of variables like SPA, then the regular Vue is a better option

**Reply by BilalKhatib** [reply]
thank you man ! it's working fine with variables and functions
